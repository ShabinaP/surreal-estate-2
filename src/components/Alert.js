import React from 'react';
import PropTypes from 'prop-types';

const Alert = ({ message, success }) => {
  if (!message) return null;
  return (
    <div className={`alert alert-${success ? 'success' : 'error'}`}>
      {message}
    </div>
  );
};

Alert.defaultProps = {
  message: '',
  success: false,
};

Alert.propTypes = {
  message: PropTypes.string,
  success: PropTypes.bool,
};
export default Alert;