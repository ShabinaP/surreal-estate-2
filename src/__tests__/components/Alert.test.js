import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../../components/Alert';

describe('Alert component', () => {
  test('renders an error message if submission is unsuccessful', () => {
    const { getByText, asFragment } = render(<Alert message="Error!" />);
    expect(getByText(/Error/).textContent).toBe('Error!');
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders a successful message when submisson is successful', () => {
    const { getByText, asFragment } = render(
      <Alert message="Success!" success />
    );
    expect(getByText(/Success/).textContent).toBe('Success!');
    expect(asFragment()).toMatchSnapshot();
  });
  test('if message is empty, no message is rendered', () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });
});