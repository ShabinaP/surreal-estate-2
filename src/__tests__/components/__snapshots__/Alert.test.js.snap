// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component if message is empty, no message is rendered 1`] = `<DocumentFragment />`;

exports[`Alert component renders a successful message when submisson is successful 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!
  </div>
</DocumentFragment>
`;

exports[`Alert component renders an error message if submission is unsuccessful 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;
